### PR TITLE
Added Kinesis-streams module

### PR DESCRIPTION
**Pull request by mayank0202**
Screenshot 2024-07-15 at 7 33 26 PM

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    content {
+      stream_mode = var.stream_mode
+    }
+  }
+
+  tags = var.tags
+}
+
+resource "aws_kinesis_stream_consumer" "this" {
+  count = var.consumer_count // This allows you to dynamically control the number of Kinesis stream consumers
```
Comment (vijay-stephen) [inline main.tf]: Please use `for_each` instead of `count`

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  name       = format("%s-consumer-%s", local.stream_name, count.index)
```
Comment (vijay-stephen) [inline main.tf]: Please don't use format, its outdated.

**Review by vijay-stephen — DISMISSED** [review]

**Comment by vijay-stephen** [comment]
@tsaucier-sf Since this is a small module shall we use the same module for firehose as well ?
We need to rename the module to `terraform-aws-arc-kinesis-stream`

**Comment by mayank0202** [comment]
> @tsaucier-sf Since this is a small module shall we use the same module for firehose as well ? We need to rename the module to `terraform-aws-arc-kinesis-stream`

I think firehose one should be different so that the code doesn't go so long

**Review by tsaucier-sf — APPROVED** [review]

**Comment by tsaucier-sf** [comment]
> > @tsaucier-sf Since this is a small module shall we use the same module for firehose as well ? We need to rename the module to `terraform-aws-arc-kinesis-stream`
> 
> I think firehose one should be different so that the code doesn't go so long

Agreed.

**Pull request merged**